common/protob/messages-ethereum-eip712.proto

```diff
@@ -26,6 +26,7 @@ message EthereumSignTypedData {
     required string primary_type = 2;                          // name of the root message struct
     optional bool metamask_v4_compat = 3 [default=true];       // use MetaMask v4 (see https://github.com/MetaMask/eth-sig-util/issues/106)
     optional ethereum.EthereumDefinitions definitions = 4;     // network and/or token definitions
+    optional bytes message_hash = 5;                           // hash of the typed data to be signed (if set, device will show the message hash as part of signature confirmation)
```

Comment:
Alternative could be `optional bool show_message_hash` and remove the equality check. Flow would be the same.

Reply:
I ended up implementing this as it seems cleaner and more explicit

Reply:
I called it `message_hash_confirmation` to match https://github.com/trezor/trezor-firmware/blob/main/common%2Fprotob%2Fmessages-bitcoin.proto... just confirming you'd prefer it be called `show_message_hash`? I don't feel strongly either way

Reply:
that's not a good place to match against because `user_confirmation` is a specific name from the SLIP-19 spec. a more appropriate would be `show_display` from `GetAddress` 
_(though "show-display" is kind of like the anglo legal convention of using two terms that mean the same thing: "aiding & abetting" "assault & battery" "cease & desist" :) )_

my point is, keep it as bytes! having the hash readily available is a good property.
in this sense `show_message_hash` or `confirm_message_hash` read clearer: "show_(the following)_message_hash: 0xabcdef12465..."